src/tests/java/htsjdk/samtools/SAMFileReaderTest.java

```diff
@@ -56,9 +57,10 @@ public void variousFormatReaderTest(final String inputFile) {
     // tests for CRAM indexing
 
     @Test(dataProvider = "SmallCRAMTest")
-    public void CRAMIndexTest(final String inputFile) {
+    public void CRAMIndexTest(final String inputFile, final String referenceFile) {
         final File input = new File(TEST_DATA_DIR, inputFile);
-        final SamReader reader = SamReaderFactory.makeDefault().open(input);
+        final File reference = new File(TEST_DATA_DIR, referenceFile);
+        final SamReader reader = SamReaderFactory.makeDefault().referenceSequence(reference).open(input);
```

Comment:
This test case should take as input both a .bai-indexed cram and a .crai-indexed cram (to ensure that `hasIndex()` returns true for both). Would also be good (if possible) to test here that the index is actually usable, since the bug you fixed manifested itself as "`hasIndex()` returns true, but queries on the index return no records".